### PR TITLE
Cygwin fixes

### PR DESCRIPTION
**Pull request by bprager**
I applied the changes recommended in #285. Cygwin now compiles without complaints. Can that be merged into the branch? Thank you.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -2232,7 +2233,7 @@ fi
 dnl ============================================================================
 dnl Enable background threads if possible.
 
-if test "x${have_pthread}" = "x1" -a "x${je_cv_os_unfair_lock}" != "xyes" ; then
+if test "x${have_pthread}" = "x1" -a "x${je_cv_os_unfair_lock}" != "xyes" -a "x${is_cywgin}" = "x1" ; then
```
Comment (davidtgoldblatt) [inline configure.ac:2236]: I don't think this is the right test -- it will effectively disable background threads on non-cygwin configurations, right?

**Review by davidtgoldblatt — COMMENTED** [review]
Nit: we keep a linear commit history, and only enable the rebase-and-merge; please squash these down to one commit, and add a more descriptive commit message.

The failures look real; possible fix inline.

File: configure.ac (hunk 1 above)
Comment (bprager) [reply configure.ac:2236]: I think you are correct. I will check and correct that. Thanks.

**Comment by AriFordsham** [comment]
Why was this closed?

**Pull request closed without merging**